frontend/src/components/elements/Video/Video.tsx

```diff
@@ -72,6 +72,21 @@ export default function Video({ element, width }: VideoProps): ReactElement {
        NOTE: This part assumes the URL is already an "embed" link.
     */
   if (type === VideoProto.Type.YOUTUBE_IFRAME) {
+    /* height of IFrame element needs to be set to avoid scrolling issue: https://github.com/streamlit/streamlit/issues/5069
+       initially IFrame element has height properly setup, but after YouTube content is loaded React renders Video with width 0
+       to fix this we save the original width to localStorage key iframe-width-${YouTubeURL},
+       when React tries to render Video component with width 0, we read the initialWidth from localStorage
+       and properly setup component's width and height
+     */
+    const localStorageIFrameWidthKey = `iframe-width-${url}`
+    let initialWidth = window.localStorage.getItem(localStorageIFrameWidthKey)
+    if (width !== 0) {
+      window.localStorage.setItem(localStorageIFrameWidthKey, width.toString())
+      initialWidth = width.toString()
+    }
+    if (width === 0 && initialWidth !== undefined && initialWidth !== null) {
```

Comment:
Hmm, are you sure that it can happen that `width` is `0` here? `width` should actually always be non-zero since it is passed from the parent container (see [here](https://github.com/streamlit/streamlit/blob/891ba59e4c622f4b242707e0dabde9f924e11f22/frontend/src/components/core/Block/ElementNodeRenderer.tsx#L368) and comment [here](https://github.com/streamlit/streamlit/blob/891ba59e4c622f4b242707e0dabde9f924e11f22/frontend/src/components/core/Block/ElementNodeRenderer.tsx#L181)). If it is `0` sometimes, there might be actually something wrong with the container inside the tab 🤔 Or maybe it is set to `0` if it is part of the not-selected tabs?

Reply:
@LukasMasuch yes, I'm sure about it, I've created a branch with `console.log` statements which show the behaviour explained in the comment, see [draft-example-of-iframe-width-0](https://github.com/streamlit/streamlit/pull/5137). If you look at Console in the dubug you will see that component width becomes 0